mypy-0.982 fails

### Describe the issue
Using `mypy-0.982` I get:

```
Processing /build/mypy-0.982
  Running command Preparing metadata (pyproject.toml)
  mypyc/build.py:48: error: Skipping analyzing "setuptools": module is installed, but missing library stubs or py.typed marker
  mypyc/build.py:48: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
  mypyc/build.py:64: error: Name "Extension" already defined (possibly by an import)
```

--- a/overrides/default.nix
+++ b/overrides/default.nix
@@ -1200,6 +1200,7 @@ lib.composeManyExtensions [
         old: {
           buildInputs = (old.buildInputs or [ ]) ++ [
             self.types-typed-ast
+            self.types-setuptools
           ];
           # Compile mypy with mypyc, which makes mypy about 4 times faster. The compiled
           # version is also the default in the wheels on Pypi that include binaries.

--- a/tests/default.nix
+++ b/tests/default.nix
@@ -103,15 +103,15 @@ builtins.removeAttrs
   # Test building poetry
   inherit poetry;
 
-  # Temporarily disabled because of mypy 0.982
-  # poetry-env =
-  #   let
-  #     env = poetry2nix.mkPoetryEnv { projectDir = ../pkgs/poetry; };
-  #   in
-  #   pkgs.runCommand "poetry-env-test" { } ''
-  #     ${env}/bin/python -c 'import requests'
-  #     touch $out
-  #   '';
+  poetry-env =
+    let
+      env = poetry2nix.mkPoetryEnv { projectDir = ../pkgs/poetry; };
+    in
+    pkgs.runCommand "poetry-env-test" { } ''
+      ${env}/bin/python -c 'import requests'
+      ${env}/bin/python -c 'import mypy'
+      touch $out
+    '';
 
   dependency-groups = callTest ./dependency-groups { };
 

